engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffActor.scala

```diff
@@ -10,22 +10,23 @@ import cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheDiffQue
 import cromwell.services.metadata.CallMetadataKeys.CallCachingKeys
 import cromwell.services.metadata.MetadataService.{GetMetadataQueryAction, MetadataLookupResponse, MetadataServiceKeyLookupFailed}
 import cromwell.services.metadata._
-import cromwell.webservice.APIResponse
-import cromwell.webservice.PerRequest.RequestComplete
-import cromwell.webservice.WorkflowJsonSupport._
 import cromwell.webservice.metadata.MetadataComponent._
 import cromwell.webservice.metadata._
-import spray.http.StatusCodes
-import spray.httpx.SprayJsonSupport._
+import spray.json.JsObject
 
 import scala.language.postfixOps
 import scala.util.{Failure, Success, Try}
 
 object CallCacheDiffActor {
   private val PlaceholderMissingHashValue = MetadataPrimitive(MetadataValue("Error: there is a hash entry for this key but the value is null !"))
-  private val CallAAndBNotFoundException = new Exception("callA and callB were run on a previous version of Cromwell on which this endpoint was not supported.")
-  private val CallANotFoundException = new Exception("callA was run on a previous version of Cromwell on which this endpoint was not supported.")
-  private val CallBNotFoundException = new Exception("callB was run on a previous version of Cromwell on which this endpoint was not supported.")
+
+  final case class CachedCallNotFoundException(message: String) extends Exception {
```

Comment:
Does this appear as a 404 in the response ? I don't see how it know that it should be that.